spark/src/main/java/org/apache/iceberg/spark/data/SparkValueReaders.java

```diff
@@ -123,7 +127,12 @@ private EnumReader(List<String> symbols) {
     @Override
     public UTF8String read(Decoder decoder, Object ignore) throws IOException {
       int index = decoder.readEnum();
-      return symbols[index];
+      if (index < 0 || index >= symbols.length) {
+        LOG.error("Unable to read the symbol in the given enum as the deserialized index {} is out of bound", index);
+        return null;
```

Comment:
My concern is that it may not be accepted upstream if Java 11 is indeed not supported. I would skip it and ask to use Java 8. You can also try to find the answer to which Java is supported in the documentation to see what the official support is.

Reply:
Thanks for looking into this! Seems the problem is that we are on an old version of Iceberg. To minimize diversion with Apache Iceberg, I would keep it and get it for free on the upgrade. One of the programs we are running now in the group is to stick to Apache Iceberg and not diverge as much as possible.

Reply:
Done.